### PR TITLE
Invite friends (#17570)

### LINKED ISSUE #17570: Invite friends
Invite friends to status should open native share
https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=2065%3A487614&mode=design&t=3R2bXdRL0DN9qi4v-1

### PR DESCRIPTION
**Pull request by alwx**
Implements native share dialog to share the profile link from the messages screen ("Invite friends to Status" banner)
Fixes #17570

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

File: src/react_native/share.cljs [hunk 1]
```diff
@@ -3,8 +3,10 @@
     ["react-native-share" :default react-native-share]))
 
 (defn open
-  [options on-success on-error]
-  (-> ^js react-native-share
-      (.open (clj->js options))
-      (.then on-success)
-      (.catch on-error)))
+  ([options]
+   (open options nil nil))
```
Comment (J-Son89) [inline src/react_native/share.cljs]: this seems slightly "dangerous" - on-success and on-error can be nil. I think you should default them to empty functions instead? i.e `(open options #() #()))` or put some check to make sure they're functions before being called. Or maybe I'm reading it wrong 🙃

**Review by J-Son89 — APPROVED** [review]
awesome! nice to have the home page (almost) feature complete :)

**Review by ulisesmac — APPROVED** [review]

File: src/react_native/share.cljs (hunk 1 above)
Comment (alwx) [reply src/react_native/share.cljs]: It actually doesn't matter in terms of how it works but I updated it for better clarity.

**Comment by churik** [comment]
@alwx 

I believe your PR should be re-done in accordance with https://github.com/status-im/status-mobile/pull/18019


Currently when I tap on "Invite friends", I goet a link without encoded metadata, i.e. `https://status.app/u#zQ3shqgVS1x4FBrsQcvouLQEdp1uMkDu2fKKffafoeaC6Avng`

**Comment by alwx** [comment]
@churik fixed!

**Comment by churik** [comment]
@Francesca-G ready for design review

**Review by Francesca-G — COMMENTED** [review]
I'm no native speaker so correct me if I'm wrong but I think the right copy should be:
_"Connect with Francesca **on** Status"_


also, is it possible to fix the icon? It should have rounded corners and no white margin :)

Design:

Screenshot 2023-12-15 alle 16 51 06

>

Implementation:

Screenshot 2023-12-15 alle 16 51 35

**Comment by alwx** [comment]
@Francesca-G it has nothing to do with the app — this part is done in the web.

**Review by Francesca-G — APPROVED** [review]

**Pull request merged**